### ISSUE TITLE
infer the version tag from upgrade info

### ISSUE THREAD

**Issue by mkaczanowski** [ChorusOne/blazar#7]
# TL;DR
Blazar requires user to provide a (docker) version tag in order to perform the upgrade.

While in cosmos there is no easy way to detect version there are a few places a user may want to infer the version from:
1. proposal title
2. proposal data with the urls to the binaries (e.g. "https://github.com/.../releases/v1.0.0/<some hash>/binary.exe")
3. proposal human written description

It would be best if there was a uniform way of specifying upgrade tag in Cosmos SDK, but we have no control over it. So instead, I propose we give the user a chance to define a small logic to infer the version from ^^ sources. It could be done via:
1. external rpc call, say `InferVersion(title, description, updateData) -> version`
2. regex, user defines a custom regex in the config
3. wasm/lua scripts

NOTE: Even if the version tag is present today in the upgrade title - some foundations do that - it happens the upgrade version changes minutes before the upgrade. For instance, instead of `v1.0.0` you're asked to use `v1.0.0-rc1`
